src/windows/mod.rs

```diff
@@ -0,0 +1,390 @@
+    }
+}
+
+/// An iterator over Mach-O segments.
+pub struct SegmentIter<'a> {
+    sections: &'a [IMAGE_SECTION_HEADER],
+    phantom: PhantomData<&'a SharedLibrary<'a>>,
+}
+
+impl<'a> fmt::Debug for SegmentIter<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SegmentIter").finish()
+    }
+}
+
+impl<'a> Iterator for SegmentIter<'a> {
+    type Item = Segment<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.sections.is_empty() {
+            None
+        } else {
+            let section = &self.sections[0];
+            self.sections = &self.sections[1..];
+            Some(Segment {
+                section,
+                phantom: PhantomData,
+            })
+        }
+    }
+}
+
+#[repr(C)]
+struct CodeViewRecord70 {
+    signature: u32,
+    pdb_signature: GUID,
+    pdb_age: u32,
+    pdb_filename: [u8; 1],
+}
+
+/// The fallback implementation of the [SharedLibrary
+/// trait](../trait.SharedLibrary.html).
+pub struct SharedLibrary<'a> {
+    module_info: MODULEINFO,
+    module_name: OsString,
+    phantom: PhantomData<&'a SharedLibrary<'a>>,
+}
+
+impl<'a> fmt::Debug for SharedLibrary<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SharedLibrary")
+            .field("module_base", &self.module_base())
+            .field("name", &self.name())
+            .field("debug_name", &self.debug_name())
+            .field("id", &self.id())
+            .field("debug_id", &self.debug_id())
+            .finish()
+    }
+}
+
+impl<'a> SharedLibrary<'a> {
+    fn new(module_info: MODULEINFO, module_name: OsString) -> SharedLibrary<'a> {
+        SharedLibrary {
+            module_info,
+            module_name,
+            phantom: PhantomData,
+        }
+    }
+
+    #[inline]
+    fn module_base(&self) -> *const c_char {
+        self.module_info.lpBaseOfDll as *const c_char
+    }
+
+    fn dos_header(&self) -> Option<&IMAGE_DOS_HEADER> {
+        let header: &IMAGE_DOS_HEADER = unsafe { mem::transmute(self.module_base()) };
+        if header.e_magic == IMAGE_DOS_SIGNATURE {
+            Some(header)
+        } else {
+            None
+        }
+    }
+
+    fn nt_headers(&self) -> Option<&IMAGE_NT_HEADERS> {
+        self.dos_header().and_then(|dos_header| {
+            let nt_headers: &IMAGE_NT_HEADERS =
+                unsafe { mem::transmute(self.module_base().offset(dos_header.e_lfanew as isize)) };
+            if nt_headers.Signature == IMAGE_NT_SIGNATURE {
+                Some(nt_headers)
+            } else {
+                println!("NOT FOUND {:x}", nt_headers.Signature);
+                None
+            }
+        })
+    }
+
+    fn codeview_record70(&self) -> Option<&CodeViewRecord70> {
+        let bias = self.virtual_memory_bias().0;
+        unsafe {
+            let debug_dictionary: *const IMAGE_DEBUG_DIRECTORY =
+                mem::transmute(self.module_base().offset(bias));
```

Comment:
This doesn't seem like a valid use of bias. This should be getting the .debug RVA from the data directory and adding that to the `module_base`. No need to include bias because by definition RVA is relative to `module_base`.